src/hooks/use-wallets-visible.ts

```diff
@@ -1,20 +1,18 @@
 import {useEffect, useState} from 'react';
 
+import {autorun} from 'mobx';
+
 import {Wallet} from '@app/models/wallet';
 
 export const useWalletsVisible = () => {
-  const [visible, setVisible] = useState(Wallet.getAllVisible().snapshot());
+  const [visible, setVisible] = useState(Wallet.getAllVisible());
+
   useEffect(() => {
     const wallets = Wallet.getAllVisible();
-    const sub = () => {
-      setVisible(wallets.snapshot());
-    };
-
-    wallets.addListener(sub);
 
-    return () => {
-      wallets.removeListener(sub);
-    };
+    autorun(() => {
```

Comment:
You don't need `visible` state here. Please use `return Wallet.getAllVisible()` instead.